homeassistant/components/device_tracker/__init__.py

```diff
@@ -61,6 +62,7 @@
             cv.time_period, cv.positive_timedelta
         ),
         vol.Optional(CONF_NEW_DEVICE_DEFAULTS, default={}): NEW_DEVICE_DEFAULTS_SCHEMA,
+        vol.Optional(CONF_LOCATION_NAME): cv.string,
```

Comment:
This would part of the legacy device tracker and we don't accept new features for that.

Reply:
If I understand the code than a device_tracker which has one of the following functions is using the legacy code: async_get_scanner, get_scanner, async_setup_scanner, setup_scanner. Which would mean that for example fritz, ping and upc_connect -device_tracker are still using the legacy code. Right? Is there a simple non-legacy device_tracker or some documentation how to write a non-legacy device_tracker?
Thanks!

Reply:
The non legacy device trackers requires a config entry for the integration. The base entity classes are here:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/device_tracker/config_entry.py